manifests/osfamily/debian.pp

```diff
@@ -1,121 +1,127 @@
 class puppet_agent::osfamily::debian{
   assert_private()
 
-  if getvar('::puppet_agent::manage_repo') == true {
 
-    include ::apt
-
-    if getvar('::puppet_agent::is_pe') == true {
-      $pe_server_version = pe_build_version()
-      if $::puppet_agent::source {
-        $source = $::puppet_agent::source
+  if $::puppet_agent::absolute_source {
+    # Absolute sources are expected to be actual packages (not repos)
+    # so when absolute_source is set just download the package to the
+    # system and finish with this class.
+    $source = $::puppet_agent::absolute_source
+    class { '::puppet_agent::prepare::package':
+      source => $source,
+    }
+    contain puppet_agent::prepare::package
+  } else {
+    if getvar('::puppet_agent::manage_repo') == true {
```

Comment:
Mixed feelings about putting the if there, and I'm not sure what is the "right" answer is in this case... so I'll just point out the differences.

For Debian this module effectively won't work with a repo at all unless `manage_repo = true`. Which seems a bit heavy handed (what if the user wants to use their own repo?).